### PR TITLE
Add lille 130m

### PR DESCRIPTION
**Pull request by Goekdeniz-Guelmez**
python -m mlx_lm.generate --model Desktop/lille --prompt "who is michael jackson"
==========
As of my last update in April 2023, Michael Jackson is a former American actor, born on February 12, 2013, and died on February 20, 2021. He was born and passed away on February 20, 2021. He was known for his work in the entertainment industry, particularly his roles in movies, television shows, and his work in the entertainment industry. He has been involved in various other roles, including acting, and has been involved in various other roles. He has been known
==========
Prompt: 10 tokens, 89.930 tokens-per-sec
Generation: 100 tokens, 331.787 tokens-per-sec
Peak memory: 0.546 GB

---

python -m mlx_lm.lora \
--train \
--model Desktop/lille \
--data mlx-community/wikisql \
--fine-tune-type lora \
--num-layers -1 \
--batch-size 1 \
--val-batches 1 \
--iters 100 \
--adapter-path Desktop/lille/lora
Loading pretrained model
Loading datasets
Loading Hugging Face dataset mlx-community/wikisql.
Training
Trainable parameters: 1.527% (1.942M/127.171M)
Starting training..., iters: 100
Calculating loss...: 100%|████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  4.88it/s]
Iter 1: Val loss 7.205, Val took 0.207s
Iter 10: Train loss 5.730, Learning Rate 1.000e-05, It/sec 5.120, Tokens/sec 385.008, Trained Tokens 752, Peak mem 0.824 GB
Iter 20: Train loss 4.312, Learning Rate 1.000e-05, It/sec 21.767, Tokens/sec 1865.405, Trained Tokens 1609, Peak mem 0.909 GB
Iter 30: Train loss 3.919, Learning Rate 1.000e-05, It/sec 27.589, Tokens/sec 2471.944, Trained Tokens 2505, Peak mem 1.004 GB
Iter 40: Train loss 3.932, Learning Rate 1.000e-05, It/sec 27.760, Tokens/sec 2512.250, Trained Tokens 3410, Peak mem 1.006 GB
Iter 50: Train loss 3.403, Learning Rate 1.000e-05, It/sec 28.352, Tokens/sec 2412.738, Trained Tokens 4261, Peak mem 1.006 GB
Iter 60: Train loss 3.149, Learning Rate 1.000e-05, It/sec 29.293, Tokens/sec 2363.979, Trained Tokens 5068, Peak mem 1.006 GB
Iter 70: Train loss 3.300, Learning Rate 1.000e-05, It/sec 28.744, Tokens/sec 2667.426, Trained Tokens 5996, Peak mem 1.006 GB
Iter 80: Train loss 3.225, Learning Rate 1.000e-05, It/sec 29.527, Tokens/sec 2598.375, Trained Tokens 6876, Peak mem 1.006 GB
Iter 90: Train loss 3.272, Learning Rate 1.000e-05, It/sec 28.603, Tokens/sec 2648.609, Trained Tokens 7802, Peak mem 1.006 GB
Calculating loss...: 100%|████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 52.15it/s]
Iter 100: Val loss 3.048, Val took 0.021s
Iter 100: Train loss 3.284, Learning Rate 1.000e-05, It/sec 29.194, Tokens/sec 2355.944, Trained Tokens 8609, Peak mem 1.006 GB

### CODE REVIEW

**Comment by Goekdeniz-Guelmez** [comment]
mlx-community/lille-130m-instruct-fp16
mlx-community/lille-130m-instruct-bf16
mlx-community/lille-130m-instruct-4bit
mlx-community/lille-130m-instruct-6bit
mlx-community/lille-130m-instruct-8bit

are online too.

**Comment by Goekdeniz-Guelmez** [comment]
@awni this is also mergable

**Review by awni — APPROVED** [review]
Thank you!
